addon/services/apollo.js

```diff
@@ -21,6 +22,31 @@ const {
 
 const { alias } = computed;
 
+function newDataFunc(observable, resultKey, resolve) {
+  let obj;
+  let mergedProps = { _apolloObservable: observable };
```

Comment:
Since you are exposing the observable to be used, I think the property shouldn't be named as a private one. `apolloObservable` looks more appropriate to me.

Reply:
My thought for keeping this private is that I wouldn't want the observable to show up in i.e. property iteration. Maybe a better, more explicit API would be to name this `__apolloObservable__` to make it obvious that it's a private property, and then expose it via some accessor function:

```js
import { getObservable } from 'ember-apollo-client';

export default Ember.Route.extend({
  model() {
    let result = this.apollo.query(...);
    let observable = getObservable(result);
    ...
  }
}
```

Thoughts on whether that's an improved design or if it's overkill? cc @viniciussbs @dfreeman

Reply:
I'd agree that it's odd to have the public documented solution for something start with "grab this property that starts with an underscore...". You could use `defineProperty` to explicitly make it nonenumerable, though it'll of course still show up in dev tools (albeit with a slightly faded-out key).

I personally like the `getObservable` approach a lot, since it abstracts away the exact details of how the access works and leaves you free to change it in the future.

Reply:
I kept the property named `_apolloObservable`, but now you're expected to get it as I documented in my proposal above. Is that sufficient, or should I make further changes (such as with `defineProperty`)?

Reply:
I like this approach, too. The `defineProperty` one could be used in the future if we return something like a `ApolloResult` or something like that instead of a `Ember.Object`. The `getObservable` looks like the best shot right now.